### PR TITLE
adding interface and implementation

### PR DESCRIPTION
**Pull request by t3rryw1**

### CODE REVIEW

File: src/ClientWrapper.php [hunk 1]
```diff
@@ -77,56 +90,4 @@ public function request($urlArray, $header, $data)
         }
         return $data;
     }
-
```
Comment (t3rryw1) [inline src/ClientWrapper.php:80]: If we are not using the mapper at all, there should be no reason for this right?

**Review by liang611zh — COMMENTED** [review]
The $header param is using here. Sometimes we have different headers for requests.

**Comment by t3rryw1** [comment]
The intention of this is to have a separate class to handle the token logic and leave GuestyClient worry-free and concerns on the end points.
It potentially can be re-used / referenced for any token update logic.

**Comment by t3rryw1** [comment]
> The $header param is using here. Sometimes we have different headers for requests.

I am not aware of this use case, can you elaborate? 
Also I think I need to update some `$header` with `$this->header`

File: src/ClientWrapper.php [hunk 2]
```diff
@@ -4,28 +4,41 @@
...
 {
     /**
      * @var HttpClient
      */
     private $client;
     private $logger;
+    private $header;
 
-    public function __construct($baseUrl,$logger=null)
+    public function __construct($baseUrl,$token, $logger=null)
```
Comment (t3rryw1) [inline src/ClientWrapper.php]: I think $token should be managed in ClientWrapper instead of here, right? 
@liang611zh is that what you mean?

**Pull request merged**